### PR TITLE
[Images] Flexible variants

### PR DESCRIPTION
**Pull request by marciocloudflare**

### CODE REVIEW

File: content/images/cloudflare-images/resize-images.md [hunk 1]
```diff
@@ -24,4 +24,124 @@ The **Fit** property describes how the width and height dimensions should be int
 | Crop        | Image will be shrunk and cropped to fit within the area specified by width and height. The image will not be enlarged. For images smaller than the given dimensions it is the same as `scale-down`. For images larger than the given dimensions, it is the same as `cover`. |
 | Pad         | Image will be resized (shrunk or enlarged) to be as large as possible within the given width or height while preserving the aspect ratio, and the extra area will be filled with a background color (white by default). |
 
-{{</table-wrap>}}
\ No newline at end of file
+{{</table-wrap>}}
+
+## Named Variants
```
Comment (sejoker) [inline content/images/cloudflare-images/resize-images.md]: We can either remove mentioning of named variants and just call them variants or we need to include 1-25 lines under Named variants section.

File: content/images/cloudflare-images/resize-images.md [hunk 2]
```diff
@@ -24,4 +24,124 @@ The **Fit** property describes how the width and height dimensions should be int
...
+
+Named variants allow you to specify ahead of time which images will need variants. This is defined during the upload. You can create variants with the Images API. For example:
```
Comment (sejoker) [inline content/images/cloudflare-images/resize-images.md]:
```suggestion
You can create variants with the Images API. For example:
```

File: content/images/cloudflare-images/resize-images.md [hunk 3]
```diff
@@ -24,4 +24,111 @@ The **Fit** property describes how the width and height dimensions should be int
...
+    -H "Content-Type: application/json" \
+    --data '{"flexible_variants": true}'
+```
+
+Once done it is possible to use resizing parameters on any Cloudflare Image. For example:
+
+```txt
+https://imagedelivery.net/<ACCOUNT_ID>/<IMAGE_ID/w=400,sharpen=3
+```
+
```
Comment (netgusto) [inline content/images/cloudflare-images/resize-images.md:54]: I would add this heads up:

Note that flexible variants cannot be used for images that require a signed delivery URL (see https://developers.cloudflare.com/images/cloudflare-images/serve-images/serve-private-images-using-signed-url-tokens/)

⚠️ Be aware that enabling flexible variants on your account effectively empowers anyone to get their desired versions of your images by changing the variant properties present in their delivery URLs, up to and including untransformed, full-resolution image, and including metadata if they chose so.

File: content/images/cloudflare-images/resize-images.md [hunk 4]
```diff
@@ -24,4 +24,111 @@ The **Fit** property describes how the width and height dimensions should be int
 | Crop        | Image will be shrunk and cropped to fit within the area specified by width and height. The image will not be enlarged. For images smaller than the given dimensions it is the same as `scale-down`. For images larger than the given dimensions, it is the same as `cover`. |
 | Pad         | Image will be resized (shrunk or enlarged) to be as large as possible within the given width or height while preserving the aspect ratio, and the extra area will be filled with a background color (white by default). |
 
-{{</table-wrap>}}
\ No newline at end of file
+{{</table-wrap>}}
+
```
Comment (sejoker) [inline content/images/cloudflare-images/resize-images.md:28]: You can create variants via API call like:
```
curl -X POST "https://api.cloudflare.com/client/v4/accounts/<ACCOUNT_ID>/images/v1/variants" \
-H "Authorization: Bearer <API_TOKEN>" \     
-H "Content-Type: application/json" \
     --data '{"id":"name-of-the-variant","options":{"fit":"scale-down","metadata":"none","width":1366,"height":768},"neverRequireSignedURLs":true}
```
Please refer to the full list of [link]

Comment (marciocloudflare) [reply content/images/cloudflare-images/resize-images.md:28]: Added below

**Review by sejoker — APPROVED** [review]

File: content/images/cloudflare-images/resize-images.md [hunk 5]
```diff
@@ -24,4 +24,120 @@ The **Fit** property describes how the width and height dimensions should be int
...
+    - A side (`"left"`, `"right"`, `"top"`, `"bottom"`) or coordinates specified on a scale from 0.0 (top or left) to 1.0 (bottom or right), 0.5 being the center. The X and Y coordinates are separated by lowercase x. For example, 0x1 means left and bottom, 0.5x0.5 is the center, 0.5x0.33 is a point in the top third of the image.
+
+- `sharpen=x`
+  - Specifies strength of sharpening filter. The value is a floating-point number between `0` (no sharpening) and `10` (maximum). `1` is a recommended value.
+
+- `blur=x`
+  - Blur radius between `1` (slight blur) and `250` (maximum). Be aware that you cannot use this option to reliably obscure image content, because savvy users can modify an image's URL and remove the blur option. Use Workers to control which options can be set.
+
+- `trim`
+    - In the form of `"left"`, `"right"`, `"top"`, `"bottom"`. Specified on a scale from `0;0` (top or left) to `1;0` (bottom or right), `0;5` being the center.
```
Comment (netgusto) [inline content/images/cloudflare-images/resize-images.md]:
```suggestion
    - Four numbers in pixels separated by a semicolon; in the form of `top;right;bottom;left`; ex: `20;30;20;0`
```

File: content/images/cloudflare-images/resize-images.md (hunk 3 above)
Comment (marciocloudflare) [reply content/images/cloudflare-images/resize-images.md:54]: Done

**Comment by netgusto** [comment]
lgtm

**Pull request merged**